[FLINK-30925][docs] Add docs about SQL Client remote mode (#21877)
apache/flink: docs/content/docs/dev/table/sqlClient.md

```
### Starting the SQL Client CLI

> The SQL Client scripts are also located in the binary directory of Flink. A user has two possibilities of starting the SQL Client CLI either by starting an embedded standalone process or by connecting to a remote [SQL Gateway]({{< ref "docs/dev/table/sql-gateway/overview" >}}). 

Reviewer: ```suggestion
The SQL Client scripts are also located in the binary directory of Flink. Users have two options for starting the SQL Client CLI, either by starting an embedded standalone process or by connecting to a remote [SQL Gateway]({{< ref "docs/dev/table/sql-gateway/overview" >}}).
```

```bash
./bin/sql-client.sh embedded
``` 

For gateway mode, you can start the CLI by calling:

```bash
./bin/sql-client.sh gateway --endpoint <gateway address>
```

In the gateway mode, the CLI submits the SQL to the specified remote gateway to execute statements.

> <span class="label label-danger">Note</span> SQL Client only supports to connect to the [REST Endpoint]({{< ref "docs/dev/table/sql-gateway/rest" >}}#rest-api) whose version is larger than v1.

Reviewer: ```suggestion
<span class="label label-danger">Note</span> SQL Client only supports connecting to the [REST Endpoint]({{< ref "docs/dev/table/sql-gateway/rest" >}}#rest-api) since version v2.
```
